KEP-2862: Graduate to STABLE. (#5774)
kubernetes/enhancements: keps/sig-node/2862-fine-grained-kubelet-authz/README.md

  cluster required to make on upgrade, in order to make use of the enhancement?
-->

> If workloads are using non finegrained permissions.

Reviewer: ```suggestion
If workloads are using non fine-grained permissions.
```

|Scenario| Result |
| -------|--------|
| Upgrade both kubelet and kube-apiserver so that feature gate is enabled in both. | workloads and kube-apiserver are able to reach kubelet|
| Upgrade only kubelet to enable the feature-gate | workloads and kube-apiserver are able to reach kubelet |
| Updrade only kube-apiserver to enable the feature-gate | workloads and kube-apiserver are able to reach kubelet |
| Rollback both kubelet and kube-apiserver so that feature gate is disabled in both. | workloads and kube-apiserver are able to reach kubelet|
| Rollback only kubelet to disable the feature-gate | workloads and kube-apiserver are able to reach kubelet |
| Rollback only kube-apiserver to disable the feature-gate | workloads and kube-apiserver are able to reach kubelet |

> If workloads are using finegrained permissions.

Reviewer: ```suggestion
If workloads are using fine-grained permissions.
```

|Scenario| Result |
| -------|--------|
| Upgrade both kubelet and kube-apiserver so that feature gate is enabled in both. | workloads and kube-apiserver are able to reach kubelet|
| Upgrade only kubelet to enable the feature-gate | workloads and kube-apiserver are able to reach kubelet |
| Updrade only kube-apiserver to enable the feature-gate | workloads wont be able to reach kubelet unless they revert to using coarse-grained permissions but kube-apiserver is able to reach kubelet |
| Rollback both kubelet and kube-apiserver so that feature gate is disabled in both. | workloads wont be able to reach kubelet unless they revert to using coarse-grained permissions but kube-apiserver is able to reach kubelet|

> | Rollback only kubelet to disable the feature-gate | workloads wont be able to reach kubelet unless they revert to coarse-grained permissions but kube-apiserver is able to reach kubelet |

Reviewer: ```suggestion
| Rollback only kubelet to disable the feature-gate | workloads won't be authorized by kubelet unless coarse-grained permissions are also granted, but kube-apiserver is able to reach kubelet |
```

  CRI or CNI may require updating that component before the kubelet.
-->

> If RBAC is not enabled for kubelet API then nothing changes. But if RBAC is enabled

Reviewer: ```suggestion
If webhook authorization is not enabled for the kubelet then nothing changes. But if webhook authorization is enabled
```
